app/src/main/java/nu/yona/app/ui/message/MessageStickyRecyclerAdapter.java

```diff
@@ -64,63 +71,107 @@ public MessageItemViewHolder onCreateViewHolder(ViewGroup parent, int viewType)
-					holder.profileIconTxt.setBackground(ContextCompat.getDrawable(YonaActivity.getActivity(), R.drawable.bg_small_self_round));
-				}
-			}
-			if (yonaObject.getLinks() != null && yonaObject.getLinks().getMarkRead() != null && !TextUtils.isEmpty(yonaObject.getLinks().getMarkRead().getHref()))
-			{
-				holder.messageContainer.setBackground(ContextCompat.getDrawable(activity, R.drawable.item_selected_gradient));
-			}
-			else
-			{
-				holder.messageContainer.setBackground(ContextCompat.getDrawable(activity, R.drawable.item_gradient));
-			}
-			holder.deleteMsg.setTag(yonaObject);
-			holder.messageContainer.setTag(yonaObject);
+			setUpNonSystemMessageItemHolder();
 		}
+		setUpMessageContainerBackground();
+	}
 
+	private void setUpSystemMessageItemHolder()
+	{
+		currentMessageItemHolder.img_avtar.setVisibility(View.GONE);
+		if (!TextUtils.isEmpty(currentYonaMessage.getMessage()))
+		{
+			currentMessageItemHolder.txtFooterMsg.setText(currentYonaMessage.getMessage());
+		}
+		currentMessageItemHolder.profileIconTxt.setVisibility(View.VISIBLE);
+		currentMessageItemHolder.profileIconTxt.setText(currentYonaMessage.getNickname().substring(0, 1).toUpperCase());
+		currentMessageItemHolder.profileIconTxt.setBackground(ContextCompat.getDrawable(YonaActivity.getActivity(), R.drawable.bg_small_admin_round));
+	}
+
+	private void setUpNonSystemMessageItemHolder()
+	{
+		currentMessageItemHolder.txtFooterMsg.setText(currentYonaMessage.getNickname());
+		if (currentYonaMessage.getNotificationMessageEnum() == NotificationMessageEnum.GOALCONFLICTMESSAGE_ANNOUNCED)
+		{
+			currentMessageItemHolder.img_avtar.setImageResource(R.drawable.adult_sad);
+			currentMessageItemHolder.img_avtar.setVisibility(View.VISIBLE);
+			currentMessageItemHolder.profileIconTxt.setVisibility(View.GONE);
+		}
+		else
+		{
+			setMessageListItemUserAvatar();
+		}
+	}
+
+	private void setMessageListItemUserAvatar()
+	{
+		if (currentYonaMessage.getLinks().getUserPhoto() != null)
+		{
+			Picasso.with(this.activity).load(currentYonaMessage.getLinks().getUserPhoto().getHref()).noFade().into(currentMessageItemHolder.img_avtar);
+			currentMessageItemHolder.img_avtar.setVisibility(View.VISIBLE);
+			currentMessageItemHolder.profileIconTxt.setVisibility(View.GONE);
+		}
+		else
+		{
+			currentMessageItemHolder.img_avtar.setVisibility(View.GONE);
+			currentMessageItemHolder.profileIconTxt.setVisibility(View.VISIBLE);
+			currentMessageItemHolder.profileIconTxt.setText(getMessageProfileIconText());
+			currentMessageItemHolder.profileIconTxt.setBackground(ContextCompat.getDrawable(YonaActivity.getActivity(), R.drawable.bg_small_self_round));
+		}
+	}
+
+	private String getMessageProfileIconText()
+	{
+		RegisterUser registerUser = currentYonaMessage.getEmbedded().getYonaUser();
+		User loggedInUser = getEventChangeManager().getDataState().getUser();
+		if (loggedInUser.getMobileNumber() != registerUser.getMobileNumber())
```

Comment:
I don't think this if-statement is doing the right thing. The icon should never reflect yourself. If you are the sender, the icon should be of the target. If you are the receiver, the icon should be of the sender.

Reply:
> If you are the sender, the icon should be of the target. If you are the receiver, the icon should be of the sender.

Is it possible that I am the sender(event actor) and I will get notified of the same?

Reply:
The majority of the messages in your message list is sent to you, but when having a conversation on an activity, the list also contains messages sent by yourself to someone.

So the rule to define the icon should be: if I'm the sender, show the receiver, otherwise the sender.